lib/Exception/UnexpectedStatusCodeException.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace DCG\Cinema\Exception;
+
+use Throwable;
+
+class UnexpectedStatusCodeException extends \RuntimeException
+{
+    private $statusCode;
+
+    /**
+     * @param int $statusCode
+     * @param string|null $message
+     * @param int $code
+     * @param Throwable|null $previous
+     */
+    public function __construct($statusCode, $message = null, $code = 0, $previous = null)
+    {
+        $this->statusCode = $statusCode;
+
+        if ($message === null) {
+            $message = "Unexpected response status code: {$statusCode}";
```

Comment:
Given that you can't force type of int for $statusCode arg because of interface, would generally prefer to force users with `sprintf('Unexpected response status code: %d', $statusCode)` just to be defensive.